lib/widgets/poll.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/zulip_localizations.dart';
+
+import '../api/model/submessage.dart';
+import 'content.dart';
+import 'store.dart';
+import 'text.dart';
+
+class PollWidget extends StatelessWidget {
+  const PollWidget({super.key, required this.poll});
+
+  final Poll poll;
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final theme = ContentTheme.of(context);
+
+    final textStylePollPrimary = const TextStyle(fontSize: 16)
+      .merge(weightVariableTextStyle(context, wght: 600));
+    final textStylePollSecondary = TextStyle(
+      fontSize: 16, color: theme.colorPollNames);
+
+    String getNameByUserId(int userId) =>
+        store.users[userId]?.fullName ?? zulipLocalizations.unknownUserName;
```

Comment:
Ah, when at https://github.com/zulip/zulip-flutter/pull/885#discussion_r1724086159 I mentioned a local variable instead of a method:
> It can probably just be a local variable in that build method, not a method or anything.

I was thinking of the result of `map` and `join` as a local variable.

I think that'd be a good further tweak — the widget expression below has a lot of details of layout, so it'd be good to move the more interesting computations out of it to happen above.